국가 클릭 시 국기 나오도록 구현 성공 및 Api 테스트 코드 작성 완료

amyyzzin opened the pull request:
**오늘의 업데이트** 

**AS-IS**
1. 메인페이지 지도 Google Map Js API를 통해 구현 완료
2. 지도에서 국가 클릭 시 간단한 세부 정보를 표시할 모달 뜨도록 구현 완료
3. 모달 내용 표시는 구현 중 (현재는 국기, 나라명만 표시)

**TO-BE**
1. 모달 내용 표시를 위해 필요한 open api 구현 및 테스트
2. 메인페이지에서 검색하면 해당 나라의 위치로 focus가 이동하도록 하는 기능 구현
3. 메인페이지에서 지도 밑에 공지사항이 보이도록 구현

**궁금한 점**

- 한 페이지에 여러 공공 API 정보를 조합해서 보여줘야 하는데, 모든 API 응답을 서버를 재 기동할 때마다 db에 저장하고 필요한 내용을 조회한 뒤 뿌려주는 것이 나을지 아니면 국가 표준 코드처럼 변하지 않을 정보들은 DB에 저장하고 코로나 공지사항처럼 실시간으로 변할 수 있는 내용은 db에 별도 저장하지 않고 세부 내용 조회하는 api를 호출할 때마다 별도 호출해서 받아 오는 것이 나을지 고민입니다

**자세한 내용**

- https://amyyzzin.notion.site/309303479e7642b1ae4559471b8d8f96

amyyzzin committed:
국가 클릭 시 국기 나오도록 구현 성공 및 Api 테스트 코드 작성 완료

Guiwoo commented on src/test/java/com/tistory/amyyzzin/trvl/ApiUtilTest.java:
```diff
@@ -4,17 +4,17 @@
 import com.tistory.amyyzzin.trvl.dto.StandardCodeResponseDto;
 import com.tistory.amyyzzin.trvl.util.ApiUtil;
 
+@ActiveProfiles("security")
```
아래 api 유틸 테스트 에 있어서 api 문서 혹은 정책이 바뀌어서 코드 전체적으로 수정해야하는 경우가 발생할수도 있을텐데 
그러면 당연히 null 이 와서 테스트 가 실패 할텐데, 이부분에 있어 좀더 명확하게 Dto 에 들어있는 항목 들을 테스트 한다면 
추후 수정 하실떄 빠르게 원인파악이 될것 같습니다. 물론 print 하는 부분이 있어 확인하실수 있지만요.

Guiwoo reviewed (commented):
잘보고 갑니다 ~ 화면 출력 부분 까지 하셔서 놀라웠습니다!!

포스트 컨스럭터 는 처음보고 배워가네요 , 이벤트리스너 를 이용해서 디비 초기화 ? 인서트 를 해주었는데 좋은 방법 하나 배워가네요 감사합니다~
Api 요청 관련해서 너무많은 api 요청이 동일한 ip 에서 다량으로 가게 되면 해당 사이트에서 요청을 일시적으로 막는데 이러한 부분에 대해서도 핸들링 해준다면 좋을것 같습니다.!!

고생하셨습니다

KimHyoJin commented on src/main/java/com/tistory/amyyzzin/trvl/controller/GoogleMapController.java:
```diff
@@ -3,12 +3,20 @@
 
+	private final CountryFlagService countryFlagService;
+
+	public GoogleMapController(CountryFlagService countryFlagService) {
```
요렇게 생성자로 넣어주는 부분을 @RequiredArgsConstructor annotation으로 대신할수도 있습니다! 참고삼아 말씀드려요

KimHyoJin reviewed (approved):
수고 많으셨습니다! 간단하지만 리뷰드렸습니다 :)
1주차의 프로젝트 셋업부분이 안되어있는 것 같은데 한번 확인부탁드릴게요! 


> 한 페이지에 여러 공공 API 정보를 조합해서 보여줘야 하는데, 모든 API 응답을 서버를 재 기동할 때마다 db에 저장하고 필요한 내용을 조회한 뒤 뿌려주는 것이 나을지 아니면 국가 표준 코드처럼 변하지 않을 정보들은 DB에 저장하고 코로나 공지사항처럼 실시간으로 변할 수 있는 내용은 db에 별도 저장하지 않고 세부 내용 조회하는 api를 호출할 때마다 별도 호출해서 받아 오는 것이 나을지 고민입니다


후자가 더 좋을 것 같습니다. 만약 실시간 변화가 있는 데이터를 굳이 기록할 필요가 없다면요. DB에 저장하는 것은 cost가 많이 드는 작업이기때문에 저장할 이유가 없다면(빠르게 변화해서 저장해도 데이터가 금방 무의미해진다면) 굳이 저장하지 않는 것이 좋습니다!

amyyzzin committed:
pull_request_template 적용

amyyzzin committed:
NoticeList // CountryBasicInfo → XML 수정필요 // CountryInfo

amyyzzin committed:
API DB 불러오기 완료

amyyzzin committed:
필요한 open api 구현 완료

1. XML/JSON 형식 모두 적용
2. XML 형식에 대해서는 RestTemplate을 통해 처리

amyyzzin committed:
BaseResponseDto 로 ResponseDto 정리 // API 명세서로 @LOB 처리

amyyzzin committed:
Modal right to left slide 구현완료, 모든 테스트에 데이터 안들어오거나 null값이면 에러 설정 완료

amyyzzin committed:
open api 요청 메서드를 generic 메서드로 통합 및 swagger 적용

1. open api 호출하는 코드가 공통되는 부분이 많아 generic 메서드로 통합
2. 우선, 테스트 코드를 작성하여 정상 작동하는 것을 확인하였고 service에서도 generic method를 호출하게끔 수정할 예정
3. swagger 적용 시 maven project는 gradle project와 달리 dependency를 여러개 추가해야함
